barretenberg/cpp/src/barretenberg/dsl/acir_format/multi_scalar_mul.cpp

```diff
@@ -25,21 +28,27 @@ template <typename Builder> void create_multi_scalar_mul_constraint(Builder& bui
         field_ct point_x;
         field_ct point_y;
         bool_ct infinite;
-        if (input.points[i].is_constant) {
-            point_x = field_ct(input.points[i].value);
-        } else {
-            point_x = field_ct::from_witness_index(&builder, input.points[i].index);
-        }
-        if (input.points[i + 1].is_constant) {
-            point_y = field_ct(input.points[i + 1].value);
-        } else {
-            point_y = field_ct::from_witness_index(&builder, input.points[i + 1].index);
-        }
-        if (input.points[i + 2].is_constant) {
-            infinite = bool_ct(field_ct(input.points[i + 2].value));
+
+        if (!has_valid_witness_assignments) {
+            infinite = bool_ct(true);
```

Comment:
should probably take in a witness_t

Reply:
yes we must keep the witness/constant nature of the input, I have updated the code.